pyccel/parser/syntactic.py

```diff
@@ -770,6 +757,16 @@ def _visit_FunctionDef(self, stmt):
         if 'private' in decorators.keys():
             is_private = True
 
+        returns = [i.expr for i in _atomic(body, cls=Return)]
+        assert all(len(i) == len(returns[0]) for i in returns)
+        returns = list(zip(*returns))
+        results = []
+        for i in returns:
+            if not all(i[0]==j for j in i) or not isinstance(i[0], Symbol):
```

Comment:
I am not sure I understand what this loop is doing, after you redefined `returns` as a list of tuples. Is `i` a tuple that contains all the expressions that have the same position in the return statements?

By they way, there is no need to redefine `returns` on line 762 (which is confusing), instead you can simply write
```
for i in zip(*returns):
    ...
```

Reply:
the answer is yes